mailpoet/lib/Subscription/Captcha/CaptchaRenderer.php

```diff
@@ -21,37 +21,28 @@ public function isSupported(): bool {
     return extension_loaded('gd') && function_exists('imagettftext');
   }
 
-  public function renderAudio(string $sessionId, $return = false) {
+  public function renderAudio(string $sessionId): void {
     if (!$this->isSupported()) {
-      return false;
+      return;
     }
 
+    header("Cache-Control: no-store, no-cache, must-revalidate");
+    header('Content-Type: audio/mpeg');
+
     $audioPath = Env::$assetsPath . '/audio/';
     $phrase = $this->getPhrase($sessionId);
-    $audio = null;
     foreach (str_split($phrase) as $character) {
       $file = $audioPath . strtolower($character) . '.mp3';
       if (!file_exists($file)) {
         throw new \RuntimeException("File not found.");
```

Comment:
Throwing an exception after you have outputted an audio/mpeg header may result in a browser displaying broken audio instead of a meaningful error (if error_reporting is on).

Reply:
@alex-mpoet What could we do instead? I didn't touch the lines, but if there's a better way, I can update it. That said, those files should always exist.

Reply:
You did not touch these lines, but you put HTTP headers before them. Originally headers were outputted after the possible exceptions. I guess a more correct way would be to first check the existence of all necessary files in a loop and throw an exception on failure, then print headers, then iterate over files again and print them.

Reply:
Ah, I see, addressed in https://github.com/mailpoet/mailpoet/pull/5727/commits/5f2f8dc590aec50a5abd31aa75c603c3ea27cbca.